### PR TITLE
Add whatsnew 66 page (#6736)

### PR DESCRIPTION
**Pull request by amychurchwell**
## Description

Adds whatsnew page for firefox 66.0 (/firefox/66.0/whatsnew/)

- WNP 66 go live - on or before - 3/12
- Release date - 3/19

## Issue / Bugzilla link
 #6736

## Testing
https://bedrock-demo-achurchwell.oregon-b.moz.works/en-US/firefox/66.0/whatsnew/

### CODE REVIEW

**Comment by amychurchwell** [comment]
Adding do not merge label, "WNP 66 go live - on or before - 3/12"

**Comment by amychurchwell** [comment]
@peiying2 hi Peiying, I realized that the copy did not include a CTA string for the hero when a user is already logged into fxa. I used:

It shouldn’t be hard to own your life online.
That’s why Firefox gives more power to you with every update.

Which is already used on whatsnew 63 and 65. 

(https://github.com/mozilla/bedrock/pull/6812)

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html [hunk 1]
```diff
@@ -0,0 +1,182 @@
+{# License, v. 2.0. If a copy of the MPL was not distributed with this
+  # file, You can obtain one at http://mozilla.org/MPL/2.0/. -#}
+
+  {% from "macros-protocol.html" import hero, call_out_compact with context %}
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html]: Nit: indentation in this file is a little off

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        <img src="{{ static('img/logos/mozilla/wordmark-dark.svg') }}" alt="Mozilla" width="112" height="32" class="c-page-header-logo-moz">
+      </div>
+    </header>
+
+    <section class="c-sticky-signup fxa-form-cta">
+        <div class="mzp-l-content">
+          <div class="c-sticky-signup-container">
+            <div class="c-sticky-signup-content">
+              <h2 class="c-sticky-signup-title">{{ _('Get a Firefox Account and all the benefits it unlocks. ') }}</h2>
+              <button type="button" class="sticky-dismiss">Dismiss this prompt.</button>
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html]: This is a hard-coded English string. At this stage we probably need to find an alternative that's in main.lang, such as "Close" https://github.com/mozilla-l10n/www.mozilla.org/blob/master/en-US/main.lang#L168 ?

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    {% set content_blocking_image='firefox/whatsnew_64/hero.png' if DIR == 'ltr' else 'firefox/whatsnew_64/hero-rtl.png' %}
+    {% call hero(
+     title=_('One log-in. Power and security everywhere.'),
+     class='mzp-has-image mzp-l-reverse t-wn64 show-fxa-default show-fxa-supported-signed-out',
+     image_url=content_blocking_image,
+     include_highres_image=true,
+     include_cta=True
+    ) %}
+    <p class="show-fxa-supported-signed-out show-fxa-default">
+      <a href="https://accounts.firefox.com/signup?service=sync&context=fx_desktop_v3&entrypoint=mozilla.org-fx.wnp64&utm_content=%2Ffirefox%2FWNP%2F64%2F&utm_campaign=wnp64&utm_source=wnp-64" class="mzp-c-button mzp-t-download">{{_('Create a Firefox Account') }}</a><br>
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html]: We should probably use the newly crreated `fxa_link_fragment` macro here? https://github.com/mozilla/bedrock/blob/master/bedrock/base/templates/macros.html#L304

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+      <span class="wn64-signin">{{ _('Have an account?') }} <a href="https://accounts.firefox.com/?action=email&service=sync&context=fx_desktop_v3&entrypoint=mozilla.org-fx.wnp64&utm_content=%2Ffirefox%2FWNP%2F64%2F&utm_campaign=wnp64&utm_source=wnp-64">{{ _('Log in') }}</a></span>
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html]: Also `fxa_link_fragment` here?

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+        <p>{{_('Feel like a secret agent when you send files online, with a private, encrypted link that ‘self-destructs.’')}}</p>
+        <p><a class="mzp-c-cta-link" data-app="send" href="
+          https://send.firefox.com/?utm_source=www.mozilla.org&utm_medium=referral&utm_campaign=wnp66&utm_content=try-send&entrypoint=mozilla.org-whatsnew66">{{_('Try Firefox Send')}}</a></p>
+      </div>
+
+      <div class="wn66-benefit wn66-lockbox">
+        {{ high_res_img('firefox/whatsnew_64/fxa-lockbox.png', {'alt': '', 'width': '590', 'height': '590'}) }}
+        <h3>{{_('Remember your passwords in Lockbox')}}</h3>
+        <p>{{_('Discover the web’s best content, and absorb it anytime – even offline – on any device. Pocket’s listen feature will even read any article aloud to you. Get it all from the Firefox toolbar or the Pocket app.')}}</p>
+        <p><a class="mzp-c-button wn66-benefit-link" data-app="lockbox" href="#">{{_('Get the Lockbox App')}}</a></p>
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html]: Seeing as this opens a modal it should probably be a `<button>`? If there's a good reason why it needs to be a link, we should at least add `role="button"`.

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+      </div>
+
+      <div class="wn66-benefit wn66-pocket">
+        {{ high_res_img('firefox/whatsnew_64/fxa-pocket.png', {'alt': '', 'width': '590', 'height': '590'}) }}
+        <h3>{{_('Put quality content in your Pocket')}}</h3>
+        <p>{{_('Discover the web’s best content, and absorb it anytime – even offline – on any device. Pocket’s listen feature will even read any article aloud to you. Get it all from the Firefox toolbar or the Pocket app.')}}</p>
+        <p><a class="mzp-c-button wn66-benefit-link" data-app="pocket" href="#">{{_('Get the Pocket App')}}</a></p>
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html]: `<button>`?

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+      <div class="wn66-benefit wn66-notes">
+        {{ high_res_img('firefox/whatsnew_64/fxa-sync.png', {'alt': '', 'width': '590', 'height': '590'}) }}
+        <h3>{{_('Pick up where you left off with Notes')}}</h3>
+        {% if l10n_has_tag('wnp66_2019') %}
+        <p>{{_('Your ideas and inspiration are secure and encrypted with Notes – and when you’re signed in to your Account, they sync from your desktop to your Android devices.')}}</p>
+        {% else %}
+        <p>{{_('Your ideas and inspiration are secure and encrypted with Notes – and when you’re signed into your Account, they sync from your desktop to your Android devices.')}}</p>
+        {% endif %}
+        <p><a class="mzp-c-button wn66-benefit-link" data-app="notes" href="#">{{_('Get the Notes App')}}</a></p>
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html]: `<button>`?

File: bedrock/firefox/views.py [hunk 8]
```diff
@@ -548,6 +548,8 @@ def get_template_names(self):
                 template = 'firefox/whatsnew/index.id.html'
         elif locale == 'zh-TW' and not version.startswith('64.'):
             template = 'firefox/whatsnew/index.zh-TW.html'
+        elif version.startswith('66.'):
+            template = 'firefox/whatsnew/whatsnew-fx66.html'
```
Comment (alexgibson) [inline bedrock/firefox/views.py]: Would be good to add a unit test for this template?

File: media/css/firefox/whatsnew/whatsnew-66.scss [hunk 9]
```diff
@@ -0,0 +1,457 @@
...
+    padding: 0 0 0 $spacing-2xl;
+    position: absolute;
+    right: 0;
+    top: 5px;
+    width: 0;
+    height: 40px;
+
+    &:after {
+        @include text-body-lg;
+        background: transparent url('#{$image-path}/icons/ui/close-black.svg') center center no-repeat;
```
Comment (alexgibson) [inline media/css/firefox/whatsnew/whatsnew-66.scss]: The black close icon here has contrast issues against the blue background, would using the white variation be easier to see?

File: media/js/firefox/whatsnew/whatsnew-66.js [hunk 10]
```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+    (function() {
```
Comment (alexgibson) [inline media/js/firefox/whatsnew/whatsnew-66.js]: Nit: indentation in this file is a little off

File: media/js/firefox/whatsnew/whatsnew-66.js [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+                    }
+                });
+            }
+        }
+
+        /*
+        * Sticky CTA & bottom CTA
+        */
+
+        var $stickyCTA = $(document.querySelectorAll('.c-sticky-signup'));
```
Comment (alexgibson) [inline media/js/firefox/whatsnew/whatsnew-66.js]: Not a blocker, but any reason to switch to jQuery here? (we have hopes and dreams of removing it from bedrock one day, so the less we write the better 🙂).

File: media/static-bundles.json [hunk 12]
```diff
@@ -1065,6 +1071,17 @@
       ],
       "name": "firefox_whatsnew_65_newsletter"
     },
+    {
+      "files": [
+        "js/libs/jquery.waypoints.min.js",
```
Comment (alexgibson) [inline media/static-bundles.json]: Is this file needed? (you seem to be using `position: sticky` which is wayyyy more performant 💯)

**Comment by alexgibson** [comment]
At small screen sizes, the close icon overlaps the headline:

screenshot_2019-03-08 get a firefox account keep your data private safe and synced

**Comment by alexgibson** [comment]
> It shouldn’t be hard to own your life online.
> That’s why Firefox gives more power to you with every update.

The main page title is untranslated when I pull the latest l10n lang files. This is kinda a big issue, we need to speak to Peiying to try and fix it 😕

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+     include_highres_image=true
+    ) %}
+    {% endcall %}
+
+    <div class="mzp-l-content">
+      <div class="wn66-benefit wn66-send">
+        {{ high_res_img('firefox/whatsnew_66/fxa-send.png', {'alt': '', 'width': '590', 'height': '590'}) }}
+        <h3>{{_('Take total control of the files you share ')}}</h3>
+        <p>{{_('Feel like a secret agent when you send files online, with a private, encrypted link that ‘self-destructs.’')}}</p>
+        <p><a class="mzp-c-cta-link" data-app="send" href="
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html]: Is there a reason why this is a cta-link, as opposed to the rest that are styled like a button?

Comment (alexgibson) [reply bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html]: Oh I see, it doesn't open a modal 👍

Comment (amychurchwell) [reply bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html]: Consulted with designers to see if we wanted to do something custom to make it seem less out of place. They decided to keep it protocol.

File: media/static-bundles.json (hunk 12 above)
Comment (amychurchwell) [reply media/static-bundles.json]: good catch! Forgotten file. 🙂

**Comment by amychurchwell** [comment]
> > It shouldn’t be hard to own your life online.
> > That’s why Firefox gives more power to you with every update.
> 
> The main page title is untranslated when I pull the latest l10n lang files. This is kinda a big issue, we need to speak to Peiying to try and fix it 😕

Yes. 😓 Pinged @peiying2 that I had to use these strings from whatsnew 63 and 65 for when a user is already logged in fxa.

File: bedrock/firefox/views.py (hunk 8 above)
Comment (amychurchwell) [reply bedrock/firefox/views.py]: excellent point! :P

**Comment by amychurchwell** [comment]
> @peiying2 hi Peiying, I realized that the copy did not include a CTA string for the hero when a user is already logged into fxa. I used:
> 
> It shouldn’t be hard to own your life online.
> That’s why Firefox gives more power to you with every update.
> 
> Which is already used on whatsnew 63 and 65.
> 
> (#6812)

Hi @flodolo -- I have an open question for Peiying who is still out of office. I was wondering if you would be able to help or point me in the right direction? I am needing to add these strings from the previous whats new pages (63 & 65) to WNP66.

**Comment by flodolo** [comment]
We need to add those strings to the template, and import the translation from 63 or 65. The real problem is that we provided RelEng with the list of active locales (i.e. completely translated) last week, and we're suddenly adding new strings (a locale might have 66 completely localized, and 63/65 not translated). I hope we don't have any cases.

P.S. If you don't get answer for a week, please escalate earlier. I'm going to take care of this, but it would have been easier with more time ahead.

**Comment by amychurchwell** [comment]
@flodolo Thank you, I really appreciate it. This was something that I did not catch in the original copy deck as a hole, my fault. Will definitely escalate earlier next time. Apologies.

**Comment by alexgibson** [comment]
I found a bug:

sticky-close-bug

The sticky bar close button is visible even before the element becomes sticky. If I click the close button before scrolling, the close button disappears. This then makes it impossible to close the bar once it becomes sticky 😅

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+  {% call hero(
+    title=_('One log-in. Power and security everywhere.'),
+    class='mzp-has-image mzp-l-reverse t-wn64 show-fxa-default show-fxa-supported-signed-out',
+    image_url=content_blocking_image,
+    include_highres_image=true,
+    include_cta=True
+  ) %}
+
+  <p class="show-fxa-supported-signed-out show-fxa-default">
+    <a {{ fxa_link_fragment(entrypoint='mozilla.org-whatsnew66', action='signup', utm_params={'campaign': 'wnp66', 'content': 'accounts-page-top-cta', 'medium': 'referral', 'source': 'wnp66'}) }} class="mzp-c-button mzp-t-download" data-button-name="Create account" data-link-type="button">
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html]: The utm params passed over to FxA here are different to the form above e.g. `entrypoint='mozilla.org-whatsnew66'` vs `entrypoint='whatsnew'`. Should they be consistent, or are they intentionally different?

File: bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html [hunk 15]
```diff
@@ -0,0 +1,188 @@
...
+      <p>{{_('Feel like a secret agent when you send files online, with a private, encrypted link that ‘self-destructs.’')}}</p>
+      <p><a class="mzp-c-cta-link" data-app="send" href="
+        https://send.firefox.com/?utm_source=www.mozilla.org&utm_medium=referral&utm_campaign=wnp66&utm_content=try-send&entrypoint=mozilla.org-whatsnew66">{{_('Try Firefox Send')}}</a></p>
+    </div>
+
+    <div class="wn66-benefit wn66-lockbox">
+      {{ high_res_img('firefox/whatsnew_64/fxa-lockbox.png', {'alt': '', 'width': '590', 'height': '590'}) }}
+      <h3>{{_('Remember your passwords in Lockbox')}}</h3>
+      <p>{{_('No more getting locked out of apps and websites. Lockbox secures all the passwords you’ve saved in Firefox and gives you easy access across all your iOS devices.')}}</p>
+      <p><button class="mzp-c-button wn66-benefit-link" data-app="lockbox" href="#">{{_('Get the Lockbox App')}}</button></p>
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html]: Can remove the left-over `href` attributes now that these are buttons?

Comment (alexgibson) [reply bedrock/firefox/templates/firefox/whatsnew/whatsnew-fx66.html]: Same for the occurences below.

File: media/static-bundles.json [hunk 16]
```diff
@@ -1065,6 +1071,16 @@
       ],
       "name": "firefox_whatsnew_65_newsletter"
     },
+    {
+      "files": [
+        "js/base/uitour-lib.js",
+        "js/base/mozilla-fxa.js",
+        "js/base/mozilla-fxa-init.js",
```
Comment (l-hedgehog) [inline media/static-bundles.json]: I think "js/base/mozilla-fxa-form.js" should also be included here?

The hidden "flow_id" and "flow_begin_time" field is currently left empty for everyone & the the form doesn't work as expected in Fx China repack.

Comment (alexgibson) [reply media/static-bundles.json]: Great catch @l-hedgehog, I'll open a new issue for this.

Comment (alexgibson) [reply media/static-bundles.json]: Filed https://github.com/mozilla/bedrock/issues/6945

Comment (l-hedgehog) [reply media/static-bundles.json]: I wish I noticed this before the 66 release.

Our QA also just told me the "Congrats! You’re using the latest version of Firefox." message is not localized, looks like it is not copied over from previous lang files into whatsnew_66.lang.

Comment (alexgibson) [reply media/static-bundles.json]: > Our QA also just told me the "Congrats! You’re using the latest version of Firefox." message is not localized, looks like it is not copied over from previous lang files into whatsnew_66.lang.

I'll file another issue for this, thanks @l-hedgehog

Comment (l-hedgehog) [reply media/static-bundles.json]: @alexgibson Thank you for the very quick response!

Comment (alexgibson) [reply media/static-bundles.json]: I've filed a separate issue for the untranslated string and CC'd folks https://github.com/mozilla/bedrock/issues/6947
